bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/services/dto/BoschSHCServiceState.java

```diff
@@ -27,15 +27,7 @@
  */
 public class BoschSHCServiceState {
 
-    /**
-     * gson instance to convert a class to json string and back.
-     */
-    private static final Gson GSON = new Gson();
-
-    /**
-     * Logger marked as transient to exclude the logger from JSON serialization.
-     */
-    private final transient Logger logger = LoggerFactory.getLogger(BoschSHCServiceState.class);
+    private final Logger logger = LoggerFactory.getLogger(getClass());
```

Comment:
How or why is the logger now  excluded from JSON serialization?

Reply:
The logger was excluded from serialization before using the `transient` keyword. However, this triggers the Sonar issue `Fields in non-serializable classes should not be "transient" (java:S2065)`.

The new solution solves this issue by using a central Gson instance retrieved via `GsonUtils.DEFAULT_GSON_INSTANCE`. This instance is configured to skip all fields named `logger` using an `ExclusionStrategy`. This does not only solve the Sonar issue, but also makes the code more efficient since we only need one Gson instance. Before every class that needed JSON serialization created an individual Gson instance.

Reply:
Thank you for explanation